irohad/ametsuchi/impl/rocksdb_common.hpp

```diff
@@ -0,0 +1,1218 @@
+  template <typename F, typename S, typename... Args>
+  inline auto enumerateKeysAndValues(RocksDbCommon &rdb,
+                                     F &&func,
+                                     S const &strformat,
+                                     Args &&... args) {
+    return rdb.enumerate(
+        [func{std::forward<F>(func)}](auto const &it,
+                                      auto const prefix_size) mutable {
+          auto const key = it->key();
+          return func(
+              rocksdb::Slice(
+                  key.data() + prefix_size + fmtstrings::kDelimiterSize,
+                  key.size() - prefix_size - 2ull * fmtstrings::kDelimiterSize),
+              it->value());
+        },
+        strformat,
+        std::forward<Args>(args)...);
+  }
+
+  template <typename F>
+  inline void mustNotExist(rocksdb::Status status, F &&op_formatter) {
+    if (status.IsNotFound())
+      return;
+
+    if (!status.ok())
+      throw IrohaDbError(12,
+                         fmt::format("{}. Failed with status: {}.",
+                                     std::forward<F>(op_formatter)(),
+                                     status.ToString()));
+
+    throw IrohaDbError(
+        13, fmt::format("{}. Exists.", std::forward<F>(op_formatter)()));
+  }
+
+  template <typename F>
+  inline void mustExist(rocksdb::Status status, F &&op_formatter) {
+    if (status.IsNotFound())
+      throw IrohaDbError(
+          14,
+          fmt::format("{}. Was not found.", std::forward<F>(op_formatter)()));
+
+    if (!status.ok())
+      throw IrohaDbError(15,
+                         fmt::format("{}. Failed with status: {}.",
+                                     std::forward<F>(op_formatter)(),
+                                     status.ToString()));
+  }
+
+  template <typename F>
+  inline void canExist(rocksdb::Status status, F &&op_formatter) {
+    if (status.IsNotFound() || status.ok())
+      return;
+
+    throw IrohaDbError(18,
+                       fmt::format("{}. Failed with status: {}.",
+                                   std::forward<F>(op_formatter)(),
+                                   status.ToString()));
+  }
+
+  template <kDbEntry kSc, typename F>
+  static void checkStatus(rocksdb::Status status, F &&op_formatter) {
+    if constexpr (kSc == kDbEntry::kAll || kSc == kDbEntry::kMustExist)
+      return mustExist(status, std::forward<F>(op_formatter));
+    else if constexpr (kSc == kDbEntry::kMustNotExist)
+      return mustNotExist(status, std::forward<F>(op_formatter));
+    else if constexpr (kSc == kDbEntry::kCanExist)
+      return canExist(status, std::forward<F>(op_formatter));
+    else
+      assert(!"Unexpected status check value");
+  }
+
+  template <kDbOperation kOp, kDbEntry kSc, typename F, typename... Args>
+  rocksdb::Status executeOperation(RocksDbCommon &common,
+                                   F &&op_formatter,
+                                   Args &&... args) {
```

Comment:
The user could think these args are to provide to F&&op_formatter function call.

So plese name it more presisely. And I suggest to put them before F. 
```suggestion
                                   OpArgs &&... op_args) {
```

Reply:
In c++ you can't locate parameter pack not in last position. Only in template param-list and only for functions.